packages/flutter_tools/lib/src/linux/linux_doctor.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/base/version.dart';
+
+import '../base/io.dart';
+import '../base/process_manager.dart';
+import '../doctor.dart';
+
+/// A validator that checks for GCC and the correct gtk version.
+class LinuxDoctorValidator extends DoctorValidator {
+  LinuxDoctorValidator() : super('Linux toolchain - develop for Linux desktop applications');
+
+  /// The minimum version of clang supported.
+  final Version minimumClangVersion = Version(3, 4, 0);
+
+  @override
+  Future<ValidationResult> validate() async {
+    ValidationType validationType = ValidationType.installed;
+    final List<ValidationMessage> messages = <ValidationMessage>[];
+    /// Check for some version of Clang.
+    final ProcessResult clangResult = await processManager.run(const <String>[
+      'clang++',
+      '--version',
+    ]);
+    if (clangResult.exitCode != 0) {
+      validationType = ValidationType.missing;
+      messages.add(ValidationMessage.error('clang++ is not installed.'));
+    } else {
+      final String firstLine = clangResult.stdout.split('\n').first.trim();
+      final String versionString = RegExp(r'[0-9]+\.[0-9]+\.[0-9]+').firstMatch(firstLine).group(0);
+      final Version version = Version.parse(versionString);
+      if (version >= minimumClangVersion) {
+        messages.add(ValidationMessage('clang++ $version'));
+      } else {
+        messages.add(ValidationMessage.error('clang++ $version is below minimum version of $minimumClangVersion'));
+      }
+    }
+
+    /// Check for the correct version of make
```

Comment:
s/correct version of make/make./

(I assume we'll have to tighten this check over time based on experience in the wild, but for now the comment should describe what's actually being checked.)